oraclizeAPI_0.4.25.sol

```diff
@@ -362,157 +399,1290 @@ contract usingOraclize {
+        address _address
+    )
+        oraclizeAPI
+        internal
+        returns (uint256 _queryPrice)
+    {
+        return oraclize.getPrice(_datasource, _address);
+    }
+
+    function oraclize_getPrice(
+        string memory _datasource,
+        address _address,
+        uint256 _gasLimit
+    )
+        oraclizeAPI
+        internal
+        returns (uint256 _queryPrice)
+    {
+        return oraclize.getPrice(_datasource, _gasLimit, _address);
+    }
+
+    function oraclize_getPrice(
+        byte _datasource,
+        address _address,
+        uint256 _gasLimit
+    )
+        oraclizeAPI
+        internal
+        returns (uint256 _queryPrice)
+    {
+        return oraclize.getPrice(_datasource, _gasLimit, _address);
+    }
+
+    function oraclize_getPrice(
+        byte _datasource,
+        uint256 _gasLimit,
+        uint256 _gasPrice
+    )
+        oraclizeAPI
+        internal
+        returns (uint256 _queryPrice)
+    {
+        _queryPrice = oraclize.price(
+            keccak256(
+                abi.encodePacked(
+                    _datasource,
+                    proofType_NONE
+                )
+            )
+        );
+        return _queryPrice += _gasLimit * _gasPrice;
+    }
+
+    function oraclize_getPrice(
+        byte _datasource,
+        uint256 _gasLimit,
+        uint256 _gasPrice,
+        byte _proofType
+    )
+        oraclizeAPI
+        internal
+        returns (uint256 _queryPrice)
+    {
+        _queryPrice = oraclize.price(
+            keccak256(
+                abi.encodePacked(
+                    _datasource,
+                    _proofType
+                )
+            )
+        );
+        return _queryPrice += _gasLimit * _gasPrice;
+    }
+
+    function oraclize_getPrice(
+        string memory _datasource,
+        uint256 _gasLimit,
+        uint256 _gasPrice
+    )
+        oraclizeAPI
+        internal
+        returns (uint256 _queryPrice)
+    {
+        _queryPrice = oraclize.price(
+            keccak256(
+                abi.encodePacked(
+                    _datasource,
+                    proofType_NONE
+                )
+            )
+        );
+        return _queryPrice += _gasLimit * _gasPrice;
+    }
+
+    function oraclize_getPrice(
+        string memory _datasource,
+        uint256 _gasLimit,
+        uint256 _gasPrice,
+        byte _proofType
+    )
+        oraclizeAPI
+        internal
+        returns (uint256 _queryPrice)
+    {
+        _queryPrice = oraclize.price(
+            keccak256(
+                abi.encodePacked(
+                    _datasource,
+                    _proofType
+                )
+            )
+        );
+        return _queryPrice += _gasLimit * _gasPrice;
     }
-
-    function oraclize_query(string datasource, string arg) oraclizeAPI internal returns (bytes32 id){
+    /**
+     *
+     * @notice oraclize_getPriceERC20(...) overloads follow...
+     *
+     */
+    function oraclize_getPriceERC20(
```

Comment:
view?

Reply:
These cannot be view due to `oraclizeAPI` modifiying state (as we've discussed elsewhere). Resolving convo.